numpy/core/tests/test_dtype.py

```diff
@@ -824,6 +821,29 @@ class Union(ctypes.Union):
         ))
         self.check(Union, expected)
 
+    def test_union_with_struct_endian_packed(self):
+        class Struct(ctypes.Structure):
+            _fields_ = [
+                ('one', ctypes.c_uint8),
+                ('two', ctypes.c_uint32.__ctype_le__)
+            ]
+            _pack_ = 1
+        class Union(ctypes.Union):
```

Comment:
Needs a test where the `Union` is packed too

Reply:
Added it.